apps/web/docker/Dockerfile

```diff
@@ -1,25 +1,63 @@
-FROM latitudedata/base AS installer
-
-ARG NODE_ENV=development
+FROM latitudedata/base AS base
 
 RUN turbo prune @latitude-data/web --docker
 
-FROM installer AS runner
+FROM base AS prunner
 
-COPY .gitignore .gitignore
-COPY --from=installer /app/out/full .
+# COMMON stage (installer)
+# ------------------------------------------------------
+FROM base AS installer
+COPY --from=prunner /app/out/json/ .
+COPY --from=prunner /app/out/pnpm-lock.yaml ./pnpm-lock.yaml
+COPY --from=prunner /app/out/pnpm-workspace.yaml ./pnpm-workspace.yaml
+COPY --from=prunner /app/out/full/ .
 
-RUN --mount=type=cache,id=pnpm,target=/pnpm/store \
-    if [ "$NODE_ENV" = "production" ]; then \
-        pnpm install --prod --frozen-lockfile; \
-    else \
-        pnpm install; \
-    fi
+RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install \
+  --frozen-lockfile \
+  --filter "@latitude-data/web..."
 
+# DEVELOPMENT stage
+FROM installer AS dev-runner
 WORKDIR /app/apps/web
+ENV NODE_ENV=development
+CMD pnpm dev
+
+# PRODUCTION stager
+# ------------------------------------------------------
+FROM installer AS prod-deps
+WORKDIR /app
+RUN --mount=type=cache,id=pnpm,target=/pnpm/store pnpm install \
+  --prod \
+  --frozen-lockfile
+
+FROM installer AS prod-builder
+RUN --mount=type=cache,id=pnpm,target=/pnpm/store \
+  BUILDING_CONTAINER=true \
+  pnpm turbo build --filter='@latitude-data/web...'
+
+FROM base AS prod-runner
+WORKDIR /app-prod
+
+# Don't run production as root
+RUN addgroup --system --gid 1001 nodejs
+RUN adduser --system --uid 1001 nextjs
+USER nextjs
+
+COPY --from=installer /app/apps/web/next.config.mjs .
+COPY --from=installer /app/apps/web/package.json .
+
+COPY --from=prod-deps --chown=nextjs:nodejs /app/node_modules ./node_modules
+COPY --from=prod-deps --chown=nextjs:nodejs /app/apps/web/package.json ./apps/web/package.json
 
-ENV NODE_ENV $NODE_ENV
+# Automatically leverage output traces to reduce image size
+# https://nextjs.org/docs/advanced-features/output-file-tracing
+COPY --from=prod-builder --chown=nextjs:nodejs /app/apps/web/.next/standalone ./apps
+COPY --from=prod-builder --chown=nextjs:nodejs /app/apps/web/.next/static ./apps/web/.next/static
+COPY --from=prod-builder --chown=nextjs:nodejs /app/apps/web/public ./apps/web/public
 
+ARG PORT=3000
+ENV PORT $PORT
+ENV NODE_ENV=production
 EXPOSE 3000
```

Comment:
you should expose $PORT instead of 3000 otherwise if someone sets up port to something else other than 3000 next won't be accesible from host